packages/internals/src/get-generators/generatorResolvers/prisma-client-js/check-dependencies/semverLt.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Returns true, if semver version `a` is lower than `b`
+ * Note: This obviously doesn't support the full semver spec.
+ * @param {string} a
+ * @param {string} b
+ */
+export function semverLt(a, b) {
+  const [major1, minor1, patch1] = a.split('.')
+  const [major2, minor2, patch2] = b.split('.')
```

Comment:
```suggestion
  const [major1, minor1, patch1] = a.split('.').map(Number)
  const [major2, minor2, patch2] = b.split('.').map(Number)
```

Otherwise, `'2' > '10'`

Reply:
Good catch, this is code I split/moved around (like most of the PR), I'll create a an issue to fix this and test it.

Reply:
https://github.com/prisma/prisma/issues/18751